planner/core/task.go

```diff
@@ -1248,7 +1248,7 @@ func (p *PhysicalTopN) checkOrderPropForSubIndexScan(idxCols []*expression.Colum
 				idxPos++
 				break
 			}
-			if len(constColsByCond) == 0 || idxPos > len(constColsByCond) || !constColsByCond[idxPos] {
+			if len(constColsByCond) == 0 || idxPos >= len(constColsByCond) || !constColsByCond[idxPos] {
```

Comment:
Maybe could change to
```suggestion
			if idxPos >= len(constColsByCond) || !constColsByCond[idxPos] {
```